### DISCUSSION TITLE
How I can change XMLDataSource in runtime mode? I need transmit my XML data as inputstream ub the birt engine.

### DISCUSSION THREAD

**Discussion by sbatyrov** [eclipse-birt/birt#1918] [Q&A]
I found one way. I can put property in the AppContext. But this way only for one data source. But how I can put inputstream in app context for two or more datasources?

**Comment by speckyspooky** [comment]
Could you give use little bit more content of the special reports to understand why you use inputstream.
From where is the stream generated and my be your example code to understand your current solution.

My be we can get your report as a small example report to test with it a little bit.
